streaming/redis.js

```diff
@@ -4,44 +4,110 @@ import { parseIntFromEnvValue } from './utils.js';
 
 /**
  * @typedef RedisConfiguration
- * @property {import('ioredis').RedisOptions} redisParams
- * @property {string} redisPrefix
  * @property {string|undefined} redisUrl
+ * @property {import('ioredis').RedisOptions} redisOptions
  */
 
+/**
+ *
+ * @param {NodeJS.ProcessEnv} env
+ * @returns {boolean}
+ */
+function hasSentinelConfiguration(env) {
+  return (
+    typeof env.REDIS_SENTINELS === 'string' &&
+    env.REDIS_SENTINELS.length > 0 &&
+    typeof env.REDIS_SENTINEL_MASTER === 'string' &&
+    env.REDIS_SENTINEL_MASTER.length > 0
+  );
+}
+
+/**
+ *
+ * @param {NodeJS.ProcessEnv} env
+ * @param {import('ioredis').SentinelConnectionOptions} commonOptions
+ * @returns {import('ioredis').SentinelConnectionOptions}
+ */
+function getSentinelConfiguration(env, commonOptions) {
+  const redisDatabase = parseIntFromEnvValue(env.REDIS_DB, 0, 'REDIS_DB');
+  const sentinelPort = parseIntFromEnvValue(env.REDIS_SENTINEL_PORT, 26379, 'REDIS_SENTINEL_PORT');
+
+  const sentinels = env.REDIS_SENTINELS.split(',').map((sentinel) => {
+    const [host, port] = sentinel.split(':', 2);
+
+    /** @type {import('ioredis').SentinelAddress} */
+    return {
+      host: host,
+      port: port ?? sentinelPort,
+      // Force support for both IPv6 and IPv4, by default ioredis sets this to 4,
+      // only allowing IPv4 connections:
+      // https://github.com/redis/ioredis/issues/1576
+      family: 0
+    };
+  });
+
+  return {
+    db: redisDatabase,
```

Comment:
I don't see anything saying `db` wouldn't still apply in Sentinel Mode.

Reply:
Yeah, the docs are not super-clear on this, but there needs to be a way to select a database in any case, so I agree.